0004-xbl-web-components.md

```diff
@@ -0,0 +1,75 @@
+6. `<handlers>`: Can be implemented with addEventListener or possibly a helper on the base element i.e:
+```
+helpers: {
+  "click": () => { /* bubble by default*/ }
+  "focus|capturing": () => { /* copy in xbl implementation */ }
+}
+```
+7. `[inherits]`: This is a way to declaratively copy attributes from a binding parent to content and keep them in sync, similar to the `<observes>` element. For example, this copies the feedURL from the parent into the 'value' field of the label, and also observes future changes
+`<feed feedURL="http://foo.com">
+<xul:label xbl:inherits="value=feedURL,tooltiptext=feedURL">`
+8. Scoped CSS Styling: this can be done with Shadow DOM
+9. Insertion Points: XBL and Shadow DOM appear to use similar models but different keywords. Further investigation is needed.
+10. `[role]`: Used for accessibility: `role="xul:toolbar"`
+
+### XBL features that don't map Web Components features
+
+1. `[implements]`: This is a way for elements with XBL bindings to implement an XPCOM interface.  XBL also auto generates the Qi bits for you. One example usage is that labels implement complex cropping behavior via `implements="nsIDOMXULLabelElement"`, but there are many others (panels, menus, scrollbars). There are around 39 instances of this feature, and this will need to be handled on a case-by-case basis. But we have some options:
+
+  * Write chrome-only webidl bindings for each of the individual nsiDOMXULFoo classes
+  * Replace things like `implements="nsIObserver"` with just a separate object that in fact implements it
+  * For some input elements like checkboxes, we could get rid of the xul interface since the corresponding HTML element has the same behavior already
+
+2. Extension Compat: For instance, the Tab Center test pilot addon causes tab browser to become vertical via inheritance with some changes.  This allows us to use the same tag name / markup so that it doesn’t need to re-write CSS. However, with the migration to Web Extensions this won’t be necessary in the general case.
+3. Binding via CSS: `-moz-binding` allows a XBL binding to be attached or detached to any element through CSS.  Web Components would be tied to HTML (based on the particular tag being created). This feature is used in [at least one place to swap an element to a different binding when setting a property](http://searchfox.org/mozilla-central/rev/7cc377ce3f0a569c5c9b362d589705cd4fecc0ac/toolkit/content/widgets/text.xml#47), but we can work around it.
+
+## XBL Usage in Firefox
+
+How many times are individual bindings used?  We've written a module to track how often XBL (and XUL) are used, and a modified version of the test runner that [keeps a count throughout the mochitest-browser suite](https://bugzilla.mozilla.org/show_bug.cgi?id=1376546). Here are initial results:
+
+TODO: ENTER LATEST RESULTS HERE
```

Comment:
I'll just copy the top few here and link to https://bugzilla.mozilla.org/show_bug.cgi?id=1376546 which has the latest results